paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -102,14 +139,132 @@ def mark_for_deployment(git_url, deploy_group, service, commit):
 
 def paasta_mark_for_deployment(args):
     """Wrapping mark_for_deployment"""
+    if args.verbose:
+        log.setLevel(level=logging.DEBUG)
+    else:
+        log.setLevel(level=logging.INFO)
     deploy_group = args.deploy_group
     service = args.service
     if service and service.startswith('services-'):
         service = service.split('services-', 1)[1]
     validate_service_name(service, soa_dir=args.soa_dir)
-    return mark_for_deployment(
+    ret = mark_for_deployment(
         git_url=args.git_url,
         deploy_group=deploy_group,
         service=service,
         commit=args.commit,
     )
+    if args.block:
+        try:
+            line = "Waiting for deployment of {0} to {1} complete".format(args.commit, deploy_group)
+            _log(
+                service=service,
+                line=line,
+                component='deploy',
+                level='event'
+            )
+            log.info(line)
+            wait_for_deployment(service=args.service,
+                                deploy_group=args.deploy_group,
+                                git_sha=args.commit,
+                                soa_dir=args.soa_dir,
+                                timeout=args.timeout)
+            line = "Deployment of {0} to {1} complete".format(args.commit, deploy_group)
+            _log(
+                service=service,
+                component='deploy',
+                line=line,
+                level='event'
+            )
+            log.info(line)
```

Comment:
Huh, I guess this is what I was missing. I was testing with the `FileLogWriter` class (which doesn't write to stdout) because I don't have scribe in my dev environment.